scripts/west_commands/build_helpers.py

```diff
@@ -155,10 +162,10 @@ def load_domains(path):
     if not domains_file.is_file():
         return Domains.from_yaml(f'''\
 default: app
-build_dir: {path}
+build_dir: {quote_path(path)}
```

Comment:
Why not use `"{path}"` here:

https://github.com/zephyrproject-rtos/zephyr/blob/4a3c5ed1f6f3ae3351fe48f11382c0a94e0aea02/scripts/west_commands/build.py#L271
pass absolute path from `find_build_dir`, so it will not cause yaml parser ambiguity. Quote the absolute path may cause issue if it contains escape characters like backslash in Windows.

https://github.com/zephyrproject-rtos/zephyr/blob/4a3c5ed1f6f3ae3351fe48f11382c0a94e0aea02/scripts/west_commands/run_common.py#L195
the `build_dir` here got from args, so it can be a numeric relative path like `123`. `quote_path` is only designed to process such case.